streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -424,23 +399,60 @@ public void evictWhile(final Supplier<Boolean> predicate,
         }
     }
 
+    @Override
+    public boolean hasKey(final K key) {
```

Comment:
The upstream will send us a "null" oldValue with the first update to any key. In that case, the correct view result is actually `null`, so we can't just assume `null` means we don't have the thing buffered, and we should look upstream.

Thus, I need both an explicit "hasKey" as well as a "get the prior value for this key (which might be null)"  method.